src/search/components/SearchForm.js

```diff
@@ -0,0 +1,30 @@
+import React, { PropTypes } from 'react';
+
+const propTypes = {
+  onSearch: PropTypes.func.isRequired,
+};
+
+class SearchForm extends React.Component {
+  constructor() {
+    super();
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(e) {
```

Comment:
The proposal for "class fields" is https://github.com/jeffmo/es-class-fields-and-static-properties. I feel like this was more solid at one point but it looks like if we want to use it it may not be included in a future version of JavaScript.